test/test_return.py

```diff
@@ -193,6 +193,26 @@ def test_return_location_edit(self):
             == ""
         ), 'location with empty variable'
 
+        assert 'success' in self.conf(
```

Comment:
Also, you can fold tests and source code patches into a single one.  The reason why most of the tests are designed as separate commits is that they have different authors. If the author is the same then it's ok to commit all together (it's up to you).

Reply:
Thanks, all things being equal I think I'd keep them separate, so each commit is doing one logical change.

Reply:
I realised that tests will be broken between these two commits: `Support URI $ sub-delimiters.` and `Tests: Remove some no longer invalid tests.`. This is not a good practice (for example, we use Buildbot and if it runs tests in between these patches then everyone who subscribed to its reports will get a lot of emails). So, please combine these 2 patches into a single one. You can leave the adding of the new tests as a separate patch.

Also, please use lowercase letter and past time after the ":" sign in the commit messages. Like "Tests: added ...". I would also recommend to quote dollar sign to emphasize this as a separate sign.

So, you will get the following set:
1) `Support URI $ sub-delimiters.`
2) `Tests: added tests for "$" sub-delimiters.`

Reply:
Understood. I shall make those adjustments. Thanks!.